### ISSUE TITLE
Show what changed from yesterday

### ISSUE THREAD

**Issue by kiernanmcgowan** [huboard/huboard#639]
During morning standup I find myself looking through my github issues to figure out what I did yesterday, and there really isn't a good way to answer that question.

It would be helpful if huboard was able to answer 'What did I do yesterday?' and show what happened with issues that I was assigned to. This could also be extended to 'What am I doing today?' and show my issues that are marked as 'doing'. 

Exposing all of this through slack would make my morning standup so easy `/huboard standup`!

**Comment by discorick** [comment]
I really like this idea! I am constantly jumping back and forth between issues and PR's to remember what I did  the day before on our slack standups.

**Comment by benjamincharity** [comment]
I would definitely lend a hand building a huboard slack bot 👍

**Comment by seanlinsley** [comment]
This would also be useful if it could be run retroactively, so we could see e.g. what tickets were added to a sprint in the middle of the sprint itself.

**Comment by benjamincharity** [comment]
If I had a nickel for every time a ticket magically appeared in my current sprint... :shakes fist:

**Comment by dealingwith** [comment]
I feel an opportunity for Huboard to sell a Sprint Protection™ Plan here.

FWIW, I had on my list to add a "take board snapshot" ~~hack~~ feature of some sort. Not as specifically helpful as the feature described here but could get the job done in the short term.

**Comment by drusellers** [comment]
> sell a Sprint Protection™ Plan here.

HAHAHAHA - that made my day 👍

**Comment by drusellers** [comment]
So, we would need to capture board state (or is it repo?) at a given time - called `T`. Then we would need to make another call later at `T'`. We could then do a diff of the data structure. Then it would be a matter of interpreting forwards / backwards / etc for each card and giving you a report. Assuming you want it for a 24 hour period, where do we store the `setting` ?

**Comment by seanlinsley** [comment]
I think that the GitHub API provides change history for labels, so it wouldn't require storing a ton of data on Huboard's side.

**Comment by dahlbyk** [comment]
My first thought re: "take baord snapshot" was [Datomic](http://www.datomic.com/) - check it out if you've never heard of it. Not sure if there's any reasonable way for us to make use of it, but it's cool nonetheless.

So I'm really seeing two distinct ideas here:

### 1. Visualization of Recent Activity

Some sort of visualization to answer questions like:

* What did I do yesterday?
* What do I have open or assigned to me?
* What has happened to this board since we locked the sprint?

We are still constrained by not storing issue data (tentative plan: https://github.com/huboard/hucache), but I think we can get a lot of useful data to answer these questions straight from the GitHub API. It's just a matter of manipulating the response into a useful experience.

The core team won't have bandwidth for visualization work, but we can certainly help with the API side.

> I think that the GitHub API provides change history for labels, so it wouldn't require storing a ton of data on Huboard's side.

I _believe_ the only way to get change history is through the Issue Events API, e.g. https://api.github.com/repos/huboard/huboard/issues/events has assigned/unassigned.

### 2. 

A HuBoard chat bot would be very cool, but it will be a while until the core team could pick this up. We could certainly create a repo here if anyone wants to give it a go (or run with it in your own repo), and we'll provide whatever support we can in implementing missing API endpoints.

**Comment by seanlinsley** [comment]
Yep, there are the `labeled` and `unlabeled` events listed here: https://developer.github.com/v3/issues/events/

**Comment by dahlbyk** [comment]
> Yep, there are the labeled and unlabeled events listed here: https://developer.github.com/v3/issues/events/

:+1: The challenge with Events is that we can only go back so far:

> Events support pagination, however the per_page option is unsupported. The fixed page size is 30 items. Fetching up to ten pages is supported, for a total of 300 events.
>
> Only events created within the past 90 days will be included in timelines. Events older than 90 days will not be included (even if the total number of events in the timeline is less than 300).

That said, Events are read-only so building a caching layer to save off every event we see is totally doable. If only we could replay events farther back. :disappointed:
